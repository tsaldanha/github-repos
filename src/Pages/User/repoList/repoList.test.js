import React from 'react';
import { mount, shallow } from 'enzyme';
import RepoList from './repoList';
import { Link } from 'react-router-dom';

const props = {
  list: [{
    node:{
      id: "abcde",
      name: "Teste 1"
    },
    node:{
      id: "qawsed",
      name: "Teste 2"
    },
  }]
};

const missing_props = {
  list: []
};


describe('Repository List Rendering', ()=>{
  let wrapper;

  beforeAll(()=>{
    wrapper = mount(<RepoList profile={props} />);
  });

  
  it('render with props ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

it('renders with missing Props without crashing', () => {
    const wrapper = shallow(<RepoList profile={missing_props} />);
 }); 

it('To NOT throw an Error', () => {
    expect(()=>{
      const wrapper = shallow(<RepoList />);
    }).not.toThrow();
  });
