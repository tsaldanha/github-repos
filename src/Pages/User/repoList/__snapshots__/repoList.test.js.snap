// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository List Rendering render with props  1`] = `ReactWrapper {}`;
